blockchain/types/transaction_signing.go

```diff
@@ -219,15 +236,113 @@ type Signer interface {
 	// Equal returns true if the given signer is the same as the receiver.
 	Equal(Signer) bool
 }
 
+type eip2930Signer struct{ EIP155Signer }
+
+// NewEIP2930Signer returns a signer that accepts EIP-2930 access list transactions,
+// EIP-155 replay protected transactions, and legacy transactions.
+func NewEIP2930Signer(chainId *big.Int) Signer {
+	return eip2930Signer{NewEIP155Signer(chainId)}
+}
+
+// ChainID returns the chain id.
+func (s eip2930Signer) ChainID() *big.Int {
+	return s.chainId
+}
+
+// Equal returns true if the given signer is the same as the receiver.
+func (s eip2930Signer) Equal(s2 Signer) bool {
+	eip2930, ok := s2.(eip2930Signer)
+	return ok && eip2930.chainId.Cmp(s.chainId) == 0
+}
+
+// Sender returns the sender address of the transaction.
+func (s eip2930Signer) Sender(tx *Transaction) (common.Address, error) {
+	if tx.Type() != TxTypeAccessList {
+		return s.EIP155Signer.Sender(tx)
+	}
+
+	if tx.ChainId().Cmp(s.chainId) != 0 {
+		return common.Address{}, ErrInvalidChainId
+	}
+
+	return tx.data.RecoverAddress(s.Hash(tx), true, func(v *big.Int) *big.Int {
+		// AL txs are defined to use 0 and 1 as their recovery
+		// id, add 27 to become equivalent to unprotected Homestead signatures.
+		V := new(big.Int).Add(v, big.NewInt(27))
+		return V
+	})
+}
+
+// SenderPubkey returns the public key derived from tx signature and txhash.
+func (s eip2930Signer) SenderPubkey(tx *Transaction) ([]*ecdsa.PublicKey, error) {
+	if tx.Type() != TxTypeAccessList {
+		return s.EIP155Signer.SenderPubkey(tx)
+	}
+
+	if tx.ChainId().Cmp(s.chainId) != 0 {
+		return nil, ErrInvalidChainId
+	}
+
+	return tx.data.RecoverPubkey(s.Hash(tx), true, func(v *big.Int) *big.Int {
+		// AL txs are defined to use 0 and 1 as their recovery
+		// id, add 27 to become equivalent to unprotected Homestead signatures.
+		V := new(big.Int).Add(v, big.NewInt(27))
+		return V
+	})
+}
+
+// SenderFeePayer returns the public key derived from tx signature and txhash.
+func (s eip2930Signer) SenderFeePayer(tx *Transaction) ([]*ecdsa.PublicKey, error) {
+	//EIP-2930(Optional access list transaction) tx don't supported fee-delegation.
+	return s.EIP155Signer.SenderFeePayer(tx)
+}
+
+// SignatureValues returns a new transaction with the given signature. This signature
+// needs to be in the [R || S || V] format where V is 0 or 1.
+func (s eip2930Signer) SignatureValues(tx *Transaction, sig []byte) (R, S, V *big.Int, err error) {
+	if tx.Type() != TxTypeAccessList {
+		return s.EIP155Signer.SignatureValues(tx, sig)
+	}
+
+	if tx.data.ChainId().Sign() != 0 && tx.data.ChainId().Cmp(s.ChainID()) != 0 {
+		return nil, nil, nil, ErrInvalidChainId
+	}
+
+	R, S, _ = decodeSignature(sig)
+	V = big.NewInt(int64(sig[64]))
+
+	return R, S, V, nil
+}
+
+// Hash returns the hash to be signed by the sender.
+// It does not uniquely identify the transaction.
+func (s eip2930Signer) Hash(tx *Transaction) common.Hash {
+	if tx.Type() != TxTypeAccessList {
```

Comment:
```suggestion
	if !tx.IsEthTypedTransaction() {
```
How about this?

Reply:
eip2930Signer only deal with AccessList tx. 
However `IsEthTypedTransaction()` function returns whether or not it is a typed transaction of Ethereum. 
The Ethereum typed transaction includes not only AccessList Tx but also Dynamic fee Tx.